code/datums/abilities/flockdrone.dm

```diff
@@ -0,0 +1,26 @@
+/datum/abilityHolder/critter/flockdrone
+	//this is jank, but we want to keep the resources stored on the drone not the abilityholder so we do this
+	var/mob/living/critter/flock/drone/flockdrone
+	var/resources_last = -1
+
+	New()
+		. = ..()
+		if (istype(owner,/mob/living/critter/flock/drone))
+			flockdrone = owner
+		else
+			throw EXCEPTION("Flockdrone abilityHolder initialized on non-flockdrone [src]")
```

Comment:
```suggestion
			stack_trace("Flockdrone abilityHolder initialized on non-flockdrone [src]")
```

doesn't really matter but `stack_trace` is standard
also throwing exceptions/`CRASH`ing in `New()` is a Bad Time. Fine here because it's right at the end, but still good to avoid

Reply:
oh also, include the ref after so we can inspect if this happens live (added to suggestion now)